Add external-dns integration (#4340)
DataDog/integrations-core: external_dns/README.md

# External-dns Integration

## Overview

> Get metrics from external-dns service in real time to:

Reviewer: ```suggestion
Get metrics from the external DNS service in real time to visualize and monitor DNS metrics collected with the Kubernetes external DNS Prometheus add on.
```

> * Visualize and monitor dns metrics collected via Kubernetes' external-dns addon

Reviewer: ```suggestion
```

>   through Prometheus

Reviewer: ```suggestion
```

> See https://github.com/kubernetes-incubator/external-dns for

Reviewer: ```suggestion
For more information about external DNS, see the [Github repo][7].
```

> more informations about external-dns

Reviewer: ```suggestion
```

## Setup
### Installation

> The external-dns check is included in the [Datadog Agent][1] package, so you don't need to install anything else on your servers.

Reviewer: ```suggestion
The external DNS check is included in the [Datadog Agent][1] package, so you don't need to install anything else on your servers.
```

### Configuration

> Edit the `external_dns.d/conf.yaml` file, in the `conf.d/` folder at the root of your [Agent's configuration directory][2], to point to your server and port, set the masters to monitor. See the [sample external_dns.d/conf.yaml][3] for all available configuration options.

Reviewer: ```suggestion
Edit the `external_dns.d/conf.yaml` file, in the `conf.d/` folder at the root of your [Agent's configuration directory][2], to point to your server and port, and to set the masters to monitor. See the [sample external_dns.d/conf.yaml][3] for all available configuration options.
```

#### Using with service discovery

> If you are using 1 dd-agent pod per kubernetes worker nodes, you could use the

Reviewer: ```suggestion
If you are using one Datadog Agent pod per Kubernetes worker node, use these example annotations on your external-dns pod to retrieve the data automatically:
```

> following annotations on your external-dns pod to get the data retrieve

Reviewer: ```suggestion
```

> automatically.

Reviewer: ```suggestion
```

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    service-discovery.datadoghq.com/external_dns.check_names: '["external_dns"]'
    service-discovery.datadoghq.com/external_dns.init_configs: '[{}]'
    service-discovery.datadoghq.com/external_dns.instances: '[[{"prometheus_url":"http://%%host%%:7979/metrics", "tags":["dns-pod:%%host%%"]}]]'
```

> **Remarks:**

Reviewer: ```suggestion
```

>  - Notice the "dns-pod" tag that will keep track of the target dns

Reviewer: ```suggestion
 - The "dns-pod" tag keeps track of the target DNS pod IP. The other tags are related to the Datadog Agent that is polling the information using the service discovery.
```

   pod IP. The other tags will be related to the dd-agent that is polling the
   informations using the service discovery.

>  - The service discovery annotations need to be done on the pod. In case of a deployment,

Reviewer: ```suggestion
 - The service discovery annotations are done on the pod. To deploy, add the annotations to the metadata of the template's specification.
```

>    add the annotations to the metadata of the template's spec.

Reviewer: ```suggestion
```

### Validation

[Run the Agent's `status` subcommand][4] and look for `external_dns` under the Checks section.

## Data Collected
### Metrics
See [metadata.csv][5] for a list of metrics provided by this integration.

### Events

> The External-DNS check does not include any events.

Reviewer: ```suggestion
The external DNS check does not include any events.
```

### Service Checks

> The External-DNS check does not include any service checks.

Reviewer: ```suggestion
The external DNS check does not include any service checks.
```

## Troubleshooting
Need help? Contact [Datadog support][6].

[1]: https://app.datadoghq.com/account/settings#agent
[2]: https://docs.datadoghq.com/agent/guide/agent-configuration-files/?tab=agentv6#agent-configuration-directory
[3]: https://github.com/DataDog/integrations-core/blob/master/external_dns/datadog_checks/external_dns/data/conf.yaml.example
[4]: https://docs.datadoghq.com/agent/guide/agent-commands/?tab=agentv6#agent-status-and-information
[5]: https://github.com/DataDog/integrations-core/blob/master/external_dns/metadata.csv

> [6]: https://docs.datadoghq.com/help

Reviewer: ```suggestion
[6]: https://docs.datadoghq.com/help
[7]: https://github.com/kubernetes-incubator/external-dns 
```
